From 6cd12da13b1db53195eb68cde1d8b9195eb90568 Mon Sep 17 00:00:00 2001
From: Yi Zhang <yix.x.zhang@intel.com>
Date: Sat, 12 Oct 2013 15:00:16 +0800
Subject: [PATCH 1/1] mmc: Change the enhanced area related sysfs output format

When the enhanced area feature was not enabled, the related sysfs will have
-EINVAL(-22) value, so change the sysfs output format to display the correct
value.

Change-Id: I17d8f007e07e8d4c1c4f0d40414d5235008f99a1
Signed-off-by: Yi Zhang <yix.x.zhang@intel.com>
Reviewed-by: Englund, Hakan <hakan.englund@intel.com>
Reviewed-by: Dong, Chuanxiao <chuanxiao.dong@intel.com>
Reviewed-by: Gao, Yunpeng <yunpeng.gao@intel.com>
Tested-by: Sun, Jianhua <jianhua.sun@intel.com>
Reviewed-by: Tang, Richard <richard.tang@intel.com>
---
 drivers/mmc/core/mmc.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/mmc/core/mmc.c b/drivers/mmc/core/mmc.c
index 7db0f77..307eee6 100644
--- a/drivers/mmc/core/mmc.c
+++ b/drivers/mmc/core/mmc.c
@@ -706,9 +706,10 @@ MMC_DEV_ATTR(name, "%s\n", card->cid.prod_name);
 MMC_DEV_ATTR(oemid, "0x%04x\n", card->cid.oemid);
 MMC_DEV_ATTR(prv, "0x%x\n", card->cid.prv);
 MMC_DEV_ATTR(serial, "0x%08x\n", card->cid.serial);
-MMC_DEV_ATTR(enhanced_area_offset, "%llu\n",
+MMC_DEV_ATTR(enhanced_area_offset, "%lld\n",
 		card->ext_csd.enhanced_area_offset);
-MMC_DEV_ATTR(enhanced_area_size, "%u\n", card->ext_csd.enhanced_area_size);
+MMC_DEV_ATTR(enhanced_area_size, "%d KBytes\n",
+		card->ext_csd.enhanced_area_size);
 MMC_DEV_ATTR(raw_rpmb_size_mult, "%#x\n", card->ext_csd.raw_rpmb_size_mult);
 MMC_DEV_ATTR(rel_sectors, "%#x\n", card->ext_csd.rel_sectors);
 
-- 
1.9.1

